<a href="https://colab.research.google.com/github/JhanellaCoronado/CPE009A-OOP/blob/main/copy_of_jupyter_notebook_oop_template.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

---
<center>

**Technological Institute of the Philippines**<br>
Computer Engineering Department<br>
Quezon city Campus</center>

---
<center><h4> Hands-on Activity 2.1 Supplementary - Literals and Variables </h4></center>

---
<center>

| Course: CPE 009 | Program: BSCpE |
|--- | --- |
| **Course Title**: Object Oriented Programming | **Date Performed:** 4/6/24 |
| **Section**: CPE12S6  | **Date Submitted:** 4/6/24 |
| **Student Name**: Coronado, Jhanella Rose V.  | **Instructor's Name:** Ma'am. Sayo |
</center>

**Objective/s of the activity:**

1.   Implement literals and variables in a python program.

**Intended Learning Outcome:**

1.  Write a simple program implementing literals and variables.
2. Use comments and identify keywords from identifiers created by users.


**Discussion:**

The use of variables, constants, and literals is crucial when it comes to handling data in programs written in Python. Variables serve as containers for values that could change while a program is running. Variables can hold a variety of data types, including strings, floats, integers, and more intricate structures like dictionaries or lists, when they are assigned with the '=' operator. Constants are named in uppercase and are used to denote values that don't change while a program runs. They can be used to store configuration or mathematical settings, but they aren't expressly supported in Python. Conversely, literals express fixed values directly within the code without requiring extra processing. They include string literals (like "hello"), boolean literals (like "True," "False"), and integer literals (like 42, 3.14). These components work together to create the basis of Python programming.

**Materials and Equipment:**

1. Computer: A system that can run an Integrated Development Environment (IDE) or a Python interpreter, allowing you to write, run, and debug Python programs.

2.  Google Colab Notebook: The cloud-based Python environment allows you to create and use Python notebooks. These notebooks, like Jupyter notebooks, allow you to write and run Python code through a browser interface.

**Procedure:**

1. A teacher wants to calculate the final grade in a CpE course and want to write it in a python program. The following are the requirements:
2. PRELIM GRADE = 50% Prelim Exam + 50% Prelim Class Standing (CS)
3. PRELIM CS = 50% Hands-on activity + 30% Quiz + 20% Assignment
4. MIDTERM GRADE = 1/3 of PRELIM GRADE + 2/3 of (50% Midterm Exam + 50% Midterm Class Standing (CS))
5. MIDTERM CS = 50% Hands-on activity + 30% Quiz + 20% Assignment
6. FINAL GRADE = 1/3 of MIDTERM GRADE + 2/3 of (50% Final Exam + 50% Final Class Standing (CS))
7. FINAL CS = 50% Hands-on activity + 30% Quiz + 20% Assignment
8. HOAs, Quizzes and Assignments are inputted as average of all submissions and are out of 100%.
9. Major exams are inputted out of 100%.
10. Show the codes that successfully run the program.



In [ ]:
def compute_prelim_cs(hands_on_activity, quiz, assignment): #prelim cs information
    """Compute Prelim Class Standing."""
    prelim_cs = 0.5 * hands_on_activity + 0.3 * quiz + 0.2 * assignment
    return prelim_cs

def compute_midterm_cs(hands_on_activity, quiz, assignment): #midterms cs information
    """Compute Midterm Class Standing."""
    midterm_cs = 0.5 * hands_on_activity + 0.3 * quiz + 0.2 * assignment
    return midterm_cs

def compute_final_cs(hands_on_activity, quiz, assignment): #final cs information
    """Compute Final Class Standing."""
    final_cs = 0.5 * hands_on_activity + 0.3 * quiz + 0.2 * assignment
    return final_cs

def compute_grade(exam, class_standing): #computation of grades information
    """Compute grade using weighted average."""
    return 0.5 * exam + 0.5 * class_standing

def compute_final_grade(prelim_exam, prelim_cs, midterm_exam, midterm_cs, final_exam, final_cs): #final grades information
    """Compute final grade using specified formula."""
    prelim_grade = compute_grade(prelim_exam, prelim_cs)
    midterm_grade = compute_grade(midterm_exam, (1/3) * prelim_grade + (2/3) * midterm_cs)
    final_grade = compute_grade(final_exam, (1/3) * midterm_grade + (2/3) * final_cs)
    return final_grade

# Input grades
hands_on_activity = float(input("Average Hands-on activity score: "))
quiz = float(input("Average Quiz score: "))
assignment = float(input("Average Assignment score: "))
prelim_exam = float(input("Enter Prelim Exam score: "))
midterm_exam = float(input("Enter Midterm Exam score: "))
final_exam = float(input("Enter Final Exam score: "))

# Processing
prelim_cs = compute_prelim_cs(hands_on_activity, quiz, assignment)
midterm_cs = compute_midterm_cs(hands_on_activity, quiz, assignment)
final_cs = compute_final_cs(hands_on_activity, quiz, assignment)
final_grade = compute_final_grade(prelim_exam, prelim_cs, midterm_exam, midterm_cs, final_exam, final_cs)

# Output of grades
print("Final Grade:", final_grade)

**Supplementary Activity**:

1. Test 3 students from the program you created.
2. The program should show the name of the student, the PRELIM, MIDTERM and FINAL grades.
3. Convert the final grade into the T.I.P.s numerical grade. Please refer to the grading system.

In [ ]:
def calculate_grade(exam, hoa, quiz, assignment): #calculation of grades information
    return 0.5 * exam + 0.5 * (0.5 * hoa + 0.3 * quiz + 0.2 * assignment)

def convert_to_tip(final_grade): #tip grades information
    if final_grade >= 92:
        return "1.0"
    elif final_grade >= 88:
        return "1.25"
    elif final_grade >= 84:
        return "1.50"
    elif final_grade >= 80:
        return "1.75"
    elif final_grade >= 75:
        return "2.0"
    elif final_grade >= 70:
        return "2.25"
    elif final_grade >= 65:
        return "2.5"
    elif final_grade >= 60:
        return "2.75"
    else:
        return "5.0"

# Test grades of three students
students = [
    {"name": "Jhanella", "prelim_exam": 83, "midterm_exam": 86, "final_exam": 87, "hoa": 98, "quiz": 99, "assignment": 98},
    {"name": "Benjamin", "prelim_exam": 81, "midterm_exam": 83, "final_exam": 82, "hoa": 99, "quiz": 99, "assignment": 99},
    {"name": "Jalene", "prelim_exam": 75, "midterm_exam": 95, "final_exam": 98, "hoa": 99, "quiz": 80, "assignment": 98}
]

# Calculate and display grades of three students
for student in students:
    prelim_grade = calculate_grade(student["prelim_exam"], student["hoa"], student["quiz"], student["assignment"])
    midterm_grade = calculate_grade(student["midterm_exam"], student["hoa"], student["quiz"], student["assignment"])
    final_grade = calculate_grade(student["final_exam"], student["hoa"], student["quiz"], student["assignment"])
    tip_grade = convert_to_tip(final_grade)
    print(f"Name: {student['name']}")
    print(f"Prelim Grade: {prelim_grade:.2f}")
    print(f"Midterm Grade: {midterm_grade:.2f}")
    print(f"Final Grade: {final_grade:.2f}")
    print(f"T.I.P. Grade: {tip_grade}\n")


Name: Jhanella
Prelim Grade: 90.65
Midterm Grade: 92.15
Final Grade: 92.65
T.I.P. Grade: 1.0

Name: Benjamin
Prelim Grade: 90.00
Midterm Grade: 91.00
Final Grade: 90.50
T.I.P. Grade: 1.25

Name: Jalene
Prelim Grade: 84.05
Midterm Grade: 94.05
Final Grade: 95.55
T.I.P. Grade: 1.0




**Questions**:
1.   Supplementary Activity:

Name: Jhanella
Prelim Grade: 90.65
Midterm Grade: 92.15
Final Grade: 92.65
T.I.P. Grade: 1.0

Name: Benjamin
Prelim Grade: 90.00
Midterm Grade: 91.00
Final Grade: 90.50
T.I.P. Grade: 1.25

Name: Jalene
Prelim Grade: 84.05
Midterm Grade: 94.05
Final Grade: 95.55
T.I.P. Grade: 1.0




**Conclusion**:

In conclusion, the goal of developing Python software to calculate students' final grades in a CpE course and convert them into numerical grades for T.I.P. has been met. To establish final marks, the software successfully asks information from a wide range of assessment components, including practical exercises, tests, assignments, and exam results. It employs the provided formulas to calculate final marks and class rankings precisely, ensuring that the grading plan is maintained. This app demonstrates how Python programming can help improve educational processes through automation and correctness. It serves as an effective tool for automating grading methods.

